### ISSUE TITLE
cxxbridge can't parse modules with inner attributes

### ISSUE THREAD

**Issue by adetaylor** [dtolnay/cxx#1378]
Reproduction:

Put this code into `test.rs`:

```rust
#[cxx::bridge]
mod ffi {
    extern "Rust" {
        fn temp();
    }
}

mod c_ffi {
    #![allow(dead_code)]
    fn may_not_be_called() {}
}
```

Then:
```
cd cxx/gen/cmd
cargo run -- test.rs
```

Problem occurs:

```
error[cxxbridge]: expected square brackets
   ┌─ ../../build/rust/tests/bindgen_test/src/lib.rs:14:6
   │
14 │     #![allow(dead_code)]
   │      ^ expected square brackets

```

This seems to be because `gen/cmd/src/gen/file.rs` line 24 uses only parse_outer, as far as I can tell.

This problem is also known as https://issues.chromium.org/issues/372062067. It's not clear that inner attributes should actually be in use here, but also, cxxbridge probably shouldn't be the thing which gets upset about them.

**Comment by 2417566723** [comment]
Hello, I have a query. In the test.rs you gave, mod c_ffi is not modified with #[cxx::bridge], but as per the execution error report, the cxxbridge tool still checks mod c_ffi?

**Comment by adetaylor** [comment]
Yes. The cxx parser has to parse all the mods to check for the relevant attributes.

**Comment by 2417566723** [comment]
Thanks a lot for the quick reply!

**Comment by makai410** [comment]
It may be rare to use inner attributes when using cxx, but it will provide users with more flexibility if we introduce it.

**Comment by makai410** [comment]
Just found a related issue #903 , so parsing modules with inner attributes is indeed necessary if we intend to support `#![cxx:bridge]`.
